lib/android/src/main/java/com/airbnb/android/react/maps/AirMapView.java

```diff
@@ -270,6 +277,98 @@ public void onLayoutChange(View v, int left, int top, int right, int bottom,
     addView(attacherGroup);
   }
 
+  public void startLocationUpdates() {
+    if (hasPermissions() && context.getCurrentActivity() != null) {
+      if (locationCallback == null)
+        locationCallback = createLocationCallBack();
+      fusedLocationClient.requestLocationUpdates(createLocationRequest(), locationCallback, Looper.getMainLooper());
+    }
+  }
+
+  public void stopLocationUpdates() {
+    if (locationCallback != null)
+      fusedLocationClient.removeLocationUpdates(locationCallback);
+  }
+
+  private LocationRequest createLocationRequest() {
+    LocationRequest locationRequest = LocationRequest.create();
+    locationRequest.setInterval(100000);
+    locationRequest.setFastestInterval(20000);
+    locationRequest.setMaxWaitTime(1500000);
+    locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+
+    LocationSettingsRequest.Builder builder = new LocationSettingsRequest.Builder()
+            .addLocationRequest(locationRequest)
+            .setAlwaysShow(true);
+
+    SettingsClient client = LocationServices.getSettingsClient(context.getCurrentActivity());
+    Task<LocationSettingsResponse> task = client.checkLocationSettings(builder.build());
+
+    task.addOnSuccessListener(context.getCurrentActivity(), new OnSuccessListener<LocationSettingsResponse>() {
+      @Override
+      public void onSuccess(LocationSettingsResponse locationSettingsResponse) {
+        if (hasPermissions())
+          startLocationUpdates();
+
+      }
+    });
+    task.addOnFailureListener(context.getCurrentActivity(), new OnFailureListener() {
+      @Override
+      public void onFailure(@NonNull Exception e) {
+        int statusCode = ((ApiException) e).getStatusCode();
+        switch (statusCode) {
+          case CommonStatusCodes.RESOLUTION_REQUIRED:
+            // Location settings are not satisfied, but this can be fixed
+            // by showing the user a dialog.
+            try {
+              // Show the dialog by calling startResolutionForResult(),
+              // and check the result in onActivityResult().
+              ResolvableApiException resolvable = (ResolvableApiException) e;
+              resolvable.startResolutionForResult(context.getCurrentActivity(), REQUEST_CODE_GPS_SETTINGS);
+            } catch (IntentSender.SendIntentException sendEx) {
+              Log.e("createLocationRequest", Log.getStackTraceString(sendEx));
+            }
+            break;
+          case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+            Log.e("createLocationRequest", "We cant do anithing");
```

Comment:
:sweat_smile: `we can't do anything`*, although maybe something like `createLocationRequest() => SETTINGS_CHANGE_UNAVAILABLE` would be even better :)